src/wp-includes/block-template-utils.php

```diff
@@ -921,8 +982,9 @@ function get_block_templates( $query = array(), $template_type = 'wp_template' )
 		$wp_query_args['tax_query']['relation'] = 'AND';
 	}
 
-	if ( isset( $query['slug__in'] ) ) {
-		$wp_query_args['post_name__in'] = $query['slug__in'];
+	if ( ! empty( $query['slug__in'] ) ) {
+		$wp_query_args['post_name__in']  = $query['slug__in'];
```

Comment:
```suggestion
		$wp_query_args['post_name__in']  = array_unique( (array) $query['slug__in'] );
```

Do we needs to check unique array otherwise we get wrong `posts_per_page`. WDYT?

Reply:
If the same slug appears twice, the post will be duplicated? I don't think we need this, no. `post_name__in` becomes [part of the filtering](https://github.com/WordPress/wordpress-develop/blob/36e994b88ceb231b34bde6ae3a4023ceb0f93d4e/src/wp-includes/class-wp-query.php#L2179). I've also tried manually by adding duplicate slugs and it still returned the proper number of posts.

Reply:
In edge case if some how we have duplicate slugs in `$query['slug__in']` value then we get wrong value for the `$wp_query_args['posts_per_page'] = count( $query['slug__in'] );` so `posts_per_page` get wrong value.

Have you gotten my point? Please correct me if I miss anything. Thanks!

Reply:
The code suggestion is attached to the wrong line (`post_name__in` instead `posts_per_page`), that's why it was confusing. I've inspected the file and see that there is a line where we do that to `posts_per_page` so I've committed the suggestion in https://github.com/WordPress/wordpress-develop/pull/4097/commits/ae409e966d5751a577901ca7cc72473ada772ed9. Thanks!